test/test_tensor.py

```diff
@@ -35,6 +37,34 @@ def test_pytorch():
     for x,y in zip(test_tinygrad(), test_pytorch()):
       np.testing.assert_allclose(x, y, atol=1e-5)
 
+  def test_backward_pass_diamond_model(self):
+    def test_tinygrad():
+      u = Tensor(U_init)
+      v = Tensor(V_init)
+      w = Tensor(W_init)
```

Comment:
I don't think this is being tested on the GPU. You should add the tensor `gpu` flag.
```
x = Tensor(x_init, gpu=self.gpu)
```